src/validators/entry/index.js

```diff
@@ -11,26 +9,17 @@ export default {
 }
 
 function validateEntry(val, {config}) {
-  const context = config.context || process.cwd()
   if (isString(val)) {
-    return getEntryError(context, val)
-  } else if (isArray(val)) {
-    return map(val, entry => getEntryError(context, entry))
+    // If a string is passed, we have to assume that the entry is valid,
+    // as there are too many ways in which the referred module could be
+    // resolved (loader-strings, query-params, resolve.moduleDirectories,
+    // resolve.extensions etc.)
+    return null
+  } else if (isArray(val) || isObject(val)) {
+    return map(val, entry => validateEntry(entry, {config}))
       .filter(error => error)
-      .join(' - ')
-  } else if (isObject(val)) {
-    return map(val, entryVal => validateEntry(entryVal, {config}))
-      .filter(error => error)
-      .join(' - ')
+      .join('\n')
```

Comment:
Curious why you changed this from ` - ` to `\n`. I would worry that it would make the output confusing (which is normally one line per error/warning).